### DISCUSSION TITLE
Integer is preferable for the type of observedGeneration

### DISCUSSION THREAD

**Discussion by dulltz** [argoproj/argo-rollouts#976] [General]
At the [v0.10.0](https://github.com/argoproj/argo-rollouts/releases/tag/v0.10.0) release, `observedGeneration` became to contain a numeric value,
but the type of the field is still *String*.

```console
$ kubectl get rollout -n default myapp -o json | jq .status.observedGeneration
"2"
```

Is the Argo-rollout community planning to convert the type to *Integer*?
In my opinion, we should do it because some tools in the eco-system expect the type of `observedGeneration` is Integer.

For example, [kstatus](https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus) accepts the Integer `observedGeneration` but does not accept numeric String. https://github.com/kubernetes-sigs/cli-utils/blob/dfab5b40aa3d8cf8493abcf4aad46c177fa0a44c/pkg/kstatus/status/generic.go#L83

**Comment by gecube** [comment]
https://github.com/argoproj/argo-rollouts/issues/3402
